src/quantity.rs

```diff
@@ -0,0 +1,215 @@
+// License: MIT
+// Copyright © 2025 Frequenz Energy-as-a-Service GmbH
+
+//! This module defines various physical quantities and their operations.
+
+/// A trait for physical quantities that supports basic arithmetic operations.
+pub trait Quantity:
+    std::ops::Add<Output = Self>
+    + std::ops::Sub<Output = Self>
+    + std::ops::Mul<Percentage, Output = Self>
+    + std::ops::Mul<f32, Output = Self>
+    + std::ops::Div<f32, Output = Self>
+    + std::ops::Div<Self, Output = f32>
+    + std::cmp::PartialOrd
+    + std::fmt::Display
+    + Copy
+    + Clone
+    + std::fmt::Debug
+    + Default
+    + Sized
+    + Send
+    + Sync
+{
+    fn zero() -> Self {
+        Self::default()
+    }
```

Comment:
Does it makes sense to provide a `zero()` alias in rust? Isn't `default()` usually zero for other scalars?